lib/Dialect/FIRRTL/Transforms/LowerIntrinsics.cpp

```diff
@@ -158,24 +163,54 @@ class EICGWrapperToClockGateConverter : public IntrinsicConverter {
   using IntrinsicConverter::IntrinsicConverter;
 
   bool check() override {
-    return hasNPorts(4) || namedPort(0, "in") || namedPort(1, "test_en") ||
-           namedPort(2, "en") || namedPort(3, "out") ||
-           typedPort<ClockType>(0) || sizedPort<UIntType>(1, 1) ||
-           sizedPort<UIntType>(2, 1) || typedPort<ClockType>(3) || hasNParam(0);
+    if (!AnnotationSet(mod).empty()) {
+      mod.emitError(name)
+          << " cannot have annotations since it is an intrinsic";
+      return true;
+    }
+    if (mod.getPorts().size() == 4) {
+      return namedPort(0, "in") || namedPort(1, "test_en") ||
+             namedPort(2, "en") || namedPort(3, "out") ||
+             typedPort<ClockType>(0) || sizedPort<UIntType>(1, 1) ||
+             sizedPort<UIntType>(2, 1) || typedPort<ClockType>(3) ||
+             hasNParam(0);
+    } else if (mod.getPorts().size() == 3) {
```

Comment:
(clang-tidy re:else-if after the return)

Reply:
Good point